cloud/google/cmd/gce-controller/main.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+	"k8s.io/apiserver/pkg/util/logs"
+	"sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/cluster-controller-app"
+	clusteroptions "sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/cluster-controller-app/options"
+	"sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/machine-controller-app"
+	machineoptions "sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/machine-controller-app/options"
+	"sigs.k8s.io/cluster-api/pkg/controller/config"
+)
+
+func main() {
+
+	fs := pflag.CommandLine
+	var controllerType, machineSetupConfigsPath string
+	fs.StringVar(&controllerType, "controller", controllerType, "specify whether this should run the machine or cluster controller")
```

Comment:
I originally thought so too, but there was some discussion awhile back about how they should be 1 binary to avoid confusion version syncing issues. The 2 controllers are likely to be at least somewhat dependent on each other's functionality. If they are 1 binary then we can say we only support running the same version of the gce-controller image for each controller. 

I don't expect them to always have the same flags though, which could also lead to some confusion in the future. Even now the machinesetup flag is only required for the machine controller, it's not passed into the cluster controller.

Reply:
I hear the versioning argument, but I think it is outweighed by the confusion that will be caused by the various flags and options that apply to one and not the other. So I think your original instincts are correct. Versioning concerns can be solved by some sort of check making sure that they are the same version within a cluster. Seems like we are optimizing for the 5% case instead of the 95% case.